### DISCUSSION TITLE
Longevity and elemental functions

### DISCUSSION THREAD

**Discussion by cuathro** [standardnotes/forum#2758] [Suggestions]
**Have you read the [Longevity Statement](https://standardnotes.org/longevity)?**
Yes.

**Is your feature request related to a problem? Please describe.**
Constructive personal opinion for the StandardNotes project:

Having reread the StandardNotes longevity policy, as a user who values the entire platform developed and who above all wishes to continue using it for as long as I live.

There are elementary functions that should be urgently included. Elementary for the simplicity of taking a note in the basic "flat editor". I admit that if StandardNotes could include photos, linking notes (personal wiki), advanced markdown, dictionaries, replace text, theme customization, information sharing, folder creation within Android tags, formatted export, Fountain syntax, etcetera it would be a dream from which I would never want to wake up. However, reality and longevity are knocking at my door, and they are asking me for balance of decisions.

I understand and value the long term project vision, but I have been waiting almost 3 years for basic functions, elementary functions to ensure that it is a useful application. As you get more complex in your ideas, the notes get more complex in the same way, hence the problem of long or complex notes. It is really exhausting, exhausting, frustrating, boring, tiring to have to perform 5, 6 or even 8 processes (screen or mouse touches) to accomplish simple processes and actions. The biggest task was not even having to migrate 2392 notes one by one to StandardNotes (I know the text importer exists), but having to accept that I have to perform bureaucratic processes to include a new idea to a specific note. Accept that secure synchronization is an ordeal for the productivity of ideas. Searching for notes on Linux is a process I cried when they implemented it, even though there is no way to jump back and forth between results, but I cried again when on Android the search capability is limited to nothing. 

**Describe the solution you'd like**
As a user, I would ask that you reconsider the urgency of the following features on your to-do list as a team (I split it into two, the first is urgent and the second falls into a much lower range of importance but would ask that it be taken into consideration in the near future)

1) URGENT

a) Search function: why: note management, taking into account longevity, should include a useful native search function. Notes, considering longevity, tend to complexify their content, so do not request complex functions, but basic management functions.

- #1019 (Simple searches) 
- #1159 (Useful searches)
- #1413 (Filters and searches for longevity)

b) Undo on Mobile (Android): Virtual keyboards in mobile applications are unstable, lack functions (Control + Z, Control + Y) and making a mistake results in having to perform at least 8 steps to try to undo a typing error. It's torture to make a mistake, whether it's deleting something by mistake or pasting copied text in the wrong place, backtracking is an unfortunate and bureaucratic process.

- #1412 (Undo in Android app). 

c) Remember scroll position: In long, complex notes, adding information or interacting between notes is elementary. For every exchange having to rummage around (without a good search function), it is a tedious and bureaucratic process

- #276 (Remember scroll position for each note)

2) NOT URGENT, BUT THAT WOULD ALLOW LONGEVITY in terms of production.

a) Multiple open notes + linking notes together (personal wiki): ability to scale notes to a complex personal development space. Fully understanding that these developments are complex due to encryption and synchronization (but let me dream a bit).

- #958 (Multiple open notes)
- https://standardnotes.com/help/58/how-do-i-link-multiple-notes-together-to-create-a-sort-of-personal-wiki (Link multiple notes)
- https://standardnotes.com/help/50/can-i-collaborate-with-others-on-a-note (Collaborate with others)

**Describe alternatives you've considered**
I am not a developer, I would have loved to be able to collaborate really, and all this text could pass as a demand, but it is the opposite, it is a request from that feeling of admiration towards all this that you built, and I would love that it continues to grow. I don't know if it helps, but there is an OpenSource notes application (that doesn't encrypt or synchronize notes) but that has those basic functions in Markdown files, that theoretically don't have line count but save the cursor position, do searches inside the notes, there is the option to undo, has the ability to keep multiple notes open. It's called Markor: https://github.com/gsantner/markor . Too bad it doesn't have the power and encryption of StandardNotes, or rather, too bad StandardNotes doesn't have those elementary functions for note taking. I understand that they focus on the application to serve long term and I share a lot of that ideal, aren't search and undo such elementary functions for a note?

**Additional context**
Take it as constructive criticism for longevity.

**Comment by JaspalSuri** [comment]
Hi @khlinuxdeveloper, thanks for the detailed feedback! We appreciate the criticism and look forward to make continuous improvements to our apps where possible.
